AC3.2-MidUnit-6-Exam/DoubleVideoViewController.swift

```diff
@@ -7,25 +7,126 @@
+            let playerItem = AVPlayerItem(url: url)
+            let player = AVPlayer(playerItem: playerItem)
+            let playerLayer = AVPlayerLayer(player: player)
+            
+            // looking at http://stackoverflow.com/questions/29000251/swift-resize-an-avplayerlayer-to-match-the-bounds-of-its-parent-container
+            
+            movies.append(playerLayer)
+            
+            if self.videoContainerTop.layer.sublayers == nil && self.videoContainerBottom.layer.sublayers == nil {
+                self.videoContainerTop.layer.addSublayer(playerLayer)
+                playerLayer.frame = videoContainerTop.bounds
+                player.play()
+            } else if videoContainerBottom.layer.sublayers == nil {
+                self.videoContainerBottom.layer.addSublayer(playerLayer)
+                playerLayer.frame = videoContainerBottom.bounds
+                player.play()
+            } else {
+                let top = movies[0]
+                let bottom = movies[1]
+                
+                if top.player?.rate != 0 && bottom.player?.rate == 0 {
+                    bottom.removeFromSuperlayer()
+                    movies[1] = playerLayer
+                    self.videoContainerBottom.layer.addSublayer(playerLayer)
+                    playerLayer.frame = videoContainerBottom.bounds
+                    player.play()
+                } else if top.player?.rate == 0 && bottom.player?.rate != 0 {
+                    top.removeFromSuperlayer()
+                    movies[0] = playerLayer
+                    self.videoContainerTop.layer.addSublayer(playerLayer)
+                    playerLayer.frame = videoContainerTop.bounds
+                    player.play()
+                } else {
+                    dismiss(animated: true) {
+                    if let url = self.movieURL {
+                        let player = AVPlayer(url: url)
+                        let playerController = AVPlayerViewController()
+                        playerController.player = player
+                    }
+                }
+                
+                // taken from my emojiCard project: https://github.com/martyav/EmojiDeck/blob/master/EmojiDeck/EmojiCardViewController.swift
+                
+                let alertController = UIAlertController(title: "You can't watch a video right now.", message: "Wait until one ends first!", preferredStyle: UIAlertControllerStyle.alert)
+                let okAction = UIAlertAction(title: "OK", style: UIAlertActionStyle.default) { (result : UIAlertAction) -> Void in
+                    print("OK")
+                }
+                alertController.addAction(okAction)
+                self.present(alertController, animated: true, completion: nil)
+                
+                print("you can't watch more than two videos")
+                return
+            }
+        }
     }
+        
+        // dismissing imagePickerController
+        dismiss(animated: true) {
```

Comment:
To me, at least, this is a case where the indenting led be to think the code was not reaching this in all cases.